### PR TITLE
Issue_438: Convert Pytest-CVP Telnet Disabled test case to Vane

### PR DESCRIPTION
**Pull request by geetanjalimanegslab**
# Please include a summary of the changes
Converted test_base_services_ntp_association.py as per the vane style guide along with that updated following files.
test_definition.yaml: Updated test def with testcase NRFU5.7
test_definition.yaml.j2: Updated J2 with testcase NRFU5.7
nrfu_tests/nrfu_tests/test_security_rp_telnet_disabled.py: Added testcase for NRFU5.7
# Any specific logic/part of code you need extra attention on
 From the output of  'show management telnet' or 'show management telnet vrf <vrf name>',  Verified that the the TELNET state is disabled for all the VRFs on DUT. 
 

# Include the Issue number and link
#438 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU Testcase)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?
Unit Tests:
 
- Tested for all the VRFS Telnet state is disabled: Passed
   [report_telnet_pass_j2.docx](https://github.com/aristanetworks/vane/files/12536092/report_telnet_pass_j2.docx)

- Tested if a few VRF's Telnet is enabled:  Failed.
   [report_telnet_failed1.docx](https://github.com/aristanetworks/vane/files/12537212/report_telnet_failed1.docx)

- Tested  for all the VRFS Telnet state is disabled: Passed (With generate test definition false )
  [report_telnet_pass_without_j2.docx](https://github.com/aristanetworks/vane/files/12536099/report_telnet_pass_without_j2.docx)

- Tested  for VRFS which are not configured: Failed
  [report_telnet_vrf_not_configured.docx](https://github.com/aristanetworks/vane/files/12536101/report_telnet_vrf_not_configured.docx)

- Tested after resolving conflicts
    [report_2309061557.docx](https://github.com/aristanetworks/vane/files/12536353/report_2309061557.docx)

 
HTML reports:
[telnet.zip](https://github.com/aristanetworks/vane/files/12537216/telnet.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/master_def.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+global_dut_filter: # This stanza is optional. If it is not provided by users than all test cases should be run on each dut.
```
Comment (MaheshGSLAB) [inline nrfu_tests/master_def.yaml:1]: lets remove this master def

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        TD: Test case for verification TELNET state as disabled for all
+        the vrfs on DUT
+        Args:
+          dut(dict): Details related to the switches
+          tests_definitions(dict): Test suite and test case parameters
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"vrfs_telnet_state": {}}
+        tops.expected_output = {"vrfs_telnet_state": {}}
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:36]: lets form the actual and expected output in below format
```
vrfs:
  default:
    telnet_state: diabled
  DVT:
    telnet_state: enabled
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            TS: Running `show vrf` on DUT and collecting the all vrfs.
+            """
+            output = dut["output"][tops.show_cmd]["json"]
+            logger.info(
+                "On device %s output of %s command is:\n%s\n",
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+            self.output = f"Output of {tops.show_cmd} command is: \n{output}\n"
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:52]:
```suggestion
            self.output += f"Output of {tops.show_cmd} command is: \n{output}\n"
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            """
+            TS: For vrf default running `show management telnet` command and for others
+            `show management telnet vrf <vrf name>` command. Verifying the state of TELNET
+            on each vrf.
+            """
+
+            for vrf in output.get("vrfs"):
+                tops.expected_output["vrfs_telnet_state"].update({vrf: "disabled"})
+
+                if vrf == "default":
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:63]:
```
telnet_cmd = "show management telnet"
if vrf != "default":
  telnet_cmd = f"show management telnet vrf {vrf}"
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    telnet_cmd = f"show management telnet vrf {vrf}"
+
+                cmd_output = tops.run_show_cmds([telnet_cmd])
+                logger.info(
+                    "On device %s output of %s command is:\n%s\n",
+                    tops.dut_name,
+                    telnet_cmd,
+                    cmd_output,
+                )
+                self.output = f"Output of {telnet_cmd} command is: \n{cmd_output}\n"
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:75]:
```suggestion
                self.output += f"Output of {telnet_cmd} command is: \n{cmd_output}\n"
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+            for vrf in output.get("vrfs"):
+                tops.expected_output["vrfs_telnet_state"].update({vrf: "disabled"})
+
+                if vrf == "default":
+                    telnet_cmd = "show management telnet"
+                else:
+                    telnet_cmd = f"show management telnet vrf {vrf}"
+
+                cmd_output = tops.run_show_cmds([telnet_cmd])
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:68]: Add this command into nested try and exception block

File: pytest.ini [hunk 7]
```diff
@@ -43,6 +43,8 @@ markers =
     eos424: Validated tests with EOS 4.24
     ssh: Verify SSH version
     vane_system_tests: Verify vane functionalities
+    security: Test cases related to security functionalities
+    nrfu_test: Network Ready For Use test cases
```
Comment (MaheshGSLAB) [inline pytest.ini:47]: lets remove this pytest.ini changes

**Review by MaheshGSLAB — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_definition.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_telnet_state_for_vrfs
+    description: Test case for verification TELNET state as disabled for all the vrfs on DUT.
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_definition.yaml:4]:
```suggestion
    description: Test case to verify that Telnet state as disabled for all the VRFs on DUT.
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+                if telnet_enabled_vrfs:
+                    tops.output_msg += (
+                        "TELNET state is not disabled for following VRFs:"
+                        f" {', '.join(telnet_enabled_vrfs)}\n"
+                    )
+
+            if not_configured_vrfs:
+                tops.output_msg += (
+                    "Following VRFs are not configured on the device"
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:103]: if VRF is not configured for telent that also means telnet is disabled for that VRF so we can have only one message.
For following VRFs Telnet is enabled: A, B, C

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+                        telnet_cmd,
+                        cmd_output,
+                    )
+                    self.output += f"Output of {telnet_cmd} command is: \n{cmd_output}\n"
+                    telnet_state = cmd_output[0]["result"].get("serverState")
+                    tops.actual_output["vrfs"].update({vrf: {"telnet_state": telnet_state}})
+                except EapiError as error:
+                    if f"VRF {vrf} not found" in str(error):
+                        not_configured_vrfs.append(vrf)
+                        tops.actual_output["vrfs"][vrf] = "VRF is not configured."
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_telnet_disabled.py:79]: If telnet is not configured for VRF that mean it is disabled only so its not a failure case

**Review by MaheshGSLAB — CHANGES_REQUESTED** [review]

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"vrfs": {}}
+        tops.expected_output = {"vrfs": {}}
+        # Forming output message if test result is pass
+        tops.output_msg = "Telnet is disabled for all the VRFs on the device."
+
+        try:
+            """
+            TS: Running `show vrf` on device and collecting the all vrfs.
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_telnet_disabled.py:41]:
```suggestion
            TS: Running `show vrf` on device and collecting all the vrfs.
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+                        cmd_output,
+                    )
+                    self.output += f"Output of {telnet_cmd} command is: \n{cmd_output}\n"
+                    telnet_state = cmd_output[0]["result"].get("serverState")
+                    tops.actual_output["vrfs"].update({vrf: {"telnet_state": telnet_state}})
+                except EapiError as exception:
+                    if f"VRF {vrf} not found" in str(exception):
+                        tops.actual_output["vrfs"].update({vrf: {"telnet_state": "disabled"}})
+
+            # Forming output message if test result is pass
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_telnet_disabled.py:78]:
```suggestion
            # Forming output message if test result is fail
```

File: nrfu_tests/test_security_rp_telnet_disabled.py [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = ""
+                telnet_enabled_vrfs = []
+
+                for vrf, telnet_state in tops.expected_output["vrfs"].items():
+                    vrf_data = tops.actual_output["vrfs"][vrf]
+
+                    if isinstance(vrf_data, dict):
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_telnet_disabled.py:86]: is this needed here ?

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

**Review by vishwajeetgslab — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**